File: src/main/java/com/inigo/arch/user/infrastucture/AuthController.java
package com.inigo.arch.user.infrastucture;

import com.inigo.arch.user.CustomUserDetailsService;
import lombok.AllArgsConstructor;
import lombok.Data;
import org.springframework.http.ResponseEntity;
import org.springframework.security.authentication.AuthenticationManager;
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.UUID;

@RestController
@RequestMapping("/auth")
public class AuthController {
    private final AuthenticationManager authManager;
    private final BearerService jwtService;

    public AuthController(AuthenticationManager authManager, BearerService jwtService) {
        this.authManager = authManager;
        this.jwtService = jwtService;
    }

    @PostMapping("/login")
    public ResponseEntity<?> login(@RequestBody AuthRequest request) {
        UserAuthentication authentication = (UserAuthentication) authManager.authenticate(
                new UsernamePasswordAuthenticationToken(request.getUsername(), request.getPassword())
        );

        String token = jwtService.generateToken(authentication.getName(),
                authentication.getEmail(),
                authentication.getId(),
                authentication.getUserRole());
        return ResponseEntity.ok(new AuthResponse(token));
    }
}

@Data
class AuthRequest {
    private String username;
    private String password;
}

@Data
@AllArgsConstructor
class AuthResponse {
    private String token;
}

File: src/main/java/com/inigo/arch/user/infrastucture/AuthController.kt
package com.inigo.arch.user.infrastucture

import com.inigo.arch.user.application.Login
import com.inigo.arch.user.domain.Password
import com.inigo.arch.user.domain.Username
import jakarta.validation.Valid
import jakarta.validation.constraints.NotNull
import org.springframework.http.ResponseEntity
import org.springframework.validation.annotation.Validated
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/auth")
@Validated
class AuthController(val login: Login) {
    @PostMapping("/login")
    fun login(@Valid @RequestBody request: AuthRequest): ResponseEntity<*> {
        val token = login.execute(Username(request.username!!), Password(request.password!!))
        return ResponseEntity.ok(AuthResponse(token.value))
    }
}

class AuthRequest {
    @field:NotNull(message = "username must not be null") val username: String? = null
    @field:NotNull(message = "password must not be null") val password: String? = null
}

data class AuthResponse (val token: String)